pkg/poseidonhttp/poseidonhttp.go

```diff
@@ -40,11 +42,66 @@ func generateMetricsHandler() map[string]http.Handler {
 	return m
 }
 
-// TODO(jiaxuanzhou): generateHealthzHandler() map[string]http.Handler
-// func generateHealthzHandler() map[string]http.Handler()
+// generateHealthzHandler generates metrics handlers.
+func generateHealthzHandler(fc firmament.FirmamentSchedulerClient) map[string]http.Handler {
+	m := make(map[string]http.Handler)
+	m[PathHealth] = newHealthzHandler(func() Health { return checkHealth(fc) })
+	return m
+}
+
+// newHealthHandler handles '/healthz' requests.
+func newHealthzHandler(hfunc func() Health) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			w.Header().Set("Allow", http.MethodGet)
+			http.Error(w, "Method Not Allowed", http.StatusMethodNotAllowed)
+			return
+		}
+		h := hfunc()
+		d, _ := json.Marshal(h)
+		if h.Health != "true" {
+			http.Error(w, string(d), http.StatusServiceUnavailable)
+			return
+		}
+		w.WriteHeader(http.StatusOK)
+		w.Write(d)
+	}
+}
+
+type Health struct {
+	Health string `json:"health"`
+}
+
+// checkHealth checks the status of firmament service and generate the status of poseidon
+func checkHealth(fc firmament.FirmamentSchedulerClient) Health {
+	h := Health{Health: "true"}
+	res, err := fc.Check(context.Background(), &firmament.HealthCheckRequest{})
+	if err != nil {
+		h.Health = "false"
+	}
+
+	if res.GetStatus() == firmament.ServingStatus_SERVING {
```

Comment:
What will happen if `res.GetStatus() != firmament.ServingStatus_SERVING`? Should the variable Health be false?

Reply:
yes， h should initialized to Health{Health: "false"}